File: noRepeat.js
function solution(arr)
{
    let answer = [];
    let cnt = 0;
    for(let i = 0; i<arr.length-1; i++) {
        if(i==0) { 
            answer[cnt++] = arr[i];
            if(arr[i]!=arr[i+1]) answer[cnt++] = arr[i+1];
            continue;
        }
        if(arr[i]==arr[i+1]) continue;
        else answer[cnt++] = arr[i+1];
    }



    return answer;
}

File: noRepeat.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> a = new ArrayList<>();
        a.add(arr[0]);
        int j = 0;
        for(int i = 1; i < arr.length; i++) {
            if(a.get(j)!=arr[i]) {
                a.add(arr[i]);
                j++;
            }
        }
        int answer[] = new int[a.size()];
        for(int i = 0; i < a.size(); i++) {
            answer[i] = a.get(i);
        }
        
        return answer;
    }
}
